lib/Page.js

```diff
@@ -492,19 +525,26 @@ class Page extends EventEmitter {
     ];
 
     const mainFrame = this._frameManager.mainFrame();
-    const timeout = typeof options.timeout === 'number' ? options.timeout : this._defaultNavigationTimeout;
-    const watcher = new NavigatorWatcher(this._frameManager, mainFrame, timeout, options);
-    const navigationPromise = watcher.navigationPromise();
-    let error = await Promise.race([
-      navigate(this._client, url, referrer),
-      navigationPromise,
-    ]);
-    if (!error)
-      error = await navigationPromise;
-    watcher.cancel();
-    helper.removeEventListeners(eventListeners);
-    if (error)
-      throw error;
+    if (this._virtualTimeController) {
+      const timeout = typeof options.timeout === 'number' ? options.timeout : this._defaultNavigationVirtualTimeout;
```

Comment:
can we use `defaultNavigationTimeout` instead of `defaultNavigationVirtualTimeout` here? This will keep API lean.